### PR TITLE
Makefile: keep cmake-build as default target

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/awesomeWM/awesome/issues/2576.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,12 @@ endif
 
 BUILDDIR=build
 
-all: awesome ;
+# Run "make" in $(BUILDDIR) by default.
+# This is required to generate all files already, which should not be generated
+# with "(sudo) make install" only later.
+cmake-build: $(BUILDDIR)/Makefile
+	$(ECHO) "Building…"
+	$(MAKE) -C $(BUILDDIR)
```
Comment (actionless) [inline Makefile]: isn't it running `$(MAKE) -C $(BUILDDIR)` twice?

Comment (blueyed) [reply Makefile]: In which case?

Just `make` defaults to `make cmake-build` now, which runs `make -C build`.
`make foo` will run `make -C build foo` (due to the implicit rule at the end).

The problem before 0b36da273 was that the default was `all`, which depended on `awesome`, which depended on `cmake-build`.  So cmake-build was run (`make -C build`), then `make -C build awesome` again.

**Comment by blueyed** [comment]
I am merging this already, since it is meant to fix master.

File: Makefile (hunk 1 above)
Comment (actionless) [reply Makefile]: `cmake-build` build target depends on `$(BUILDDIR)/Makefile`

and both of them are running `$(MAKE) -C $(BUILDDIR)` one of them just bypassing args at the end (`$@`)

Comment (blueyed) [reply Makefile]: `make` (or `make cmake-build`) does not also build the `%` target.

Comment (blueyed) [reply Makefile]: `%` is only used if not other target is specified (including the default).

**Pull request merged**